src/framework/components/layout-group/component.js

```diff
@@ -0,0 +1,209 @@
+
+        // Listen to existing children being resized
+        this.entity.children.forEach(function(child) {
+            this._listenForResizeEvents(child, 'on');
+        }.bind(this));
+
+        // Listen to newly added children being resized
+        this.entity.on('childinsert', this._onChildInsert, this);
+        this.entity.on('childremove', this._onChildRemove, this);
+
+        // Listen for ElementComponents and LayoutChildComponents being added
+        // to self or to children - covers cases where they are not already
+        // present at the point when this component is constructed.
+        system.app.systems.element.on('add', this._onElementComponentAdd, this);
+        system.app.systems.element.on('beforeremove', this._onElementComponentRemove, this);
+        system.app.systems.layoutchild.on('add', this._onLayoutChildComponentAdd, this);
+        system.app.systems.layoutchild.on('beforeremove', this._onLayoutChildComponentRemove, this);
+
+        this._scheduleReflow();
+    };
+    LayoutGroupComponent = pc.inherits(LayoutGroupComponent, pc.Component);
+
+    pc.extend(LayoutGroupComponent.prototype, {
+        _isSelfOrChild: function(entity) {
+            return (entity === this.entity) || (this.entity.children.indexOf(entity) !== -1);
+        },
+
+        _listenForResizeEvents: function(target, onOff) {
+            if (target.element) {
+                target.element[onOff]('resize', this._onResize, this);
+            }
+
+            if (target.layoutchild) {
+                target.layoutchild[onOff]('resize', this._onResize, this);
+            }
+        },
+
+        _onElementComponentAdd: function(entity, component) {
+            if (this._isSelfOrChild(entity)) {
+                component.on('resize', this._onResize, this);
+            }
+        },
+
+        _onElementComponentRemove: function(entity, component) {
+            if (this._isSelfOrChild(entity)) {
+                component.off('resize', this._onResize, this);
+            }
+        },
+
+        _onLayoutChildComponentAdd: function(entity, component) {
+            if (this._isSelfOrChild(entity)) {
+                component.on('resize', this._onResize, this);
+            }
+        },
+
+        _onLayoutChildComponentRemove: function(entity, component) {
+            if (this._isSelfOrChild(entity)) {
+                component.off('resize', this._onResize, this);
+            }
+        },
+
+        _onResize: function() {
+            this._scheduleReflow();
+        },
+
+        _onChildInsert: function(child) {
+            this._listenForResizeEvents(child, 'on');
+            this._scheduleReflow();
+        },
+
+        _onChildRemove: function(child) {
+            this._listenForResizeEvents(child, 'off');
+            this._scheduleReflow();
+        },
+
+        _scheduleReflow: function() {
+            if (!this._isPerformingReflow) {
+                this.fire('schedulereflow', this);
```

Comment:
It looks like the 'schedulereflow' event is only used internally with in the layoutgroup component system. If so, it seems like using events for this adds unnecessary overhead. Can we just do this by calling into the system directly?